native/membrane_videocompositor_opengl_rust/src/scene.rs

```diff
@@ -0,0 +1,252 @@
+//! High-level structures, such as Scenes and Videos.
+
+use crate::{
+    framebuffers::{self, YUVRenderTarget},
+    shaders::ShaderProgram,
+    textures::YUVPlanarTexture,
+};
+use glad_gles2::gl;
+
+/// Describes where a video should be located in the scene space.
+/// All coordinates have to be in the range [-1, 1].
+pub struct VideoPlacementTemplate {
+    pub top_right: (f32, f32),
```

Comment:
```suggestion
    pub top_right: Vec2f,
```

I would consider introducing a struct for this tuple. Maybe even using `nalgebra` provided types, but these will probably increase compilation time a lot.

Reply:
We will probably use some OpenGL-specific maths library, which would provide functionality similar to `nalgebra` in the future. For now I'll just make a Point(f32, f32)

Reply:
👍🏻